From cedf3db7f6b542ae4709e9bb1dc41e7be7bcc00c Mon Sep 17 00:00:00 2001
From: Gerald Pfeifer <gerald@pfeifer.com>
Date: Tue, 28 Jan 2020 22:40:25 +0100
Subject: [PATCH 02/15] msi: Properly declare variables in msi/tests/utils.h as
 extern.

Omitting the extern storage qualifier when declaring a global variable in
a header file is mistake. If that header is included by several files it
results in multiple definitions of the same variable (unless -fcommon is
specified or assumed, the latter being the case for GCC 9.x and earlier).

This fixes building with GCC 10.

Signed-off-by: Gerald Pfeifer <gerald@pfeifer.com>
Signed-off-by: Hans Leidekker <hans@codeweavers.com>
Signed-off-by: Alexandre Julliard <julliard@winehq.org>
---
 dlls/msi/tests/utils.h | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/dlls/msi/tests/utils.h b/dlls/msi/tests/utils.h
index f16e2d5ff5..0880553863 100644
--- a/dlls/msi/tests/utils.h
+++ b/dlls/msi/tests/utils.h
@@ -16,12 +16,12 @@
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
  */
 
-char PROG_FILES_DIR[MAX_PATH];
-char PROG_FILES_DIR_NATIVE[MAX_PATH];
-char COMMON_FILES_DIR[MAX_PATH];
-char APP_DATA_DIR[MAX_PATH];
-char WINDOWS_DIR[MAX_PATH];
-char CURR_DIR[MAX_PATH];
+extern char PROG_FILES_DIR[MAX_PATH];
+extern char PROG_FILES_DIR_NATIVE[MAX_PATH];
+extern char COMMON_FILES_DIR[MAX_PATH];
+extern char APP_DATA_DIR[MAX_PATH];
+extern char WINDOWS_DIR[MAX_PATH];
+extern char CURR_DIR[MAX_PATH];
 
 BOOL get_system_dirs(void);
 BOOL get_user_dirs(void);
-- 
2.26.2

